cmd/starter/c/starter.c

```diff
@@ -353,16 +353,26 @@ static void set_rpc_privileges(void) {
 
     memset(priv, 0, sizeof(struct privileges));
 
-    priv->capabilities.effective |= capflag(CAP_MKNOD);
-    priv->capabilities.effective |= capflag(CAP_SETGID);
-    priv->capabilities.effective |= capflag(CAP_SETUID);
-    priv->capabilities.effective |= capflag(CAP_SYS_ADMIN);
-    priv->capabilities.effective |= capflag(CAP_SYS_CHROOT);
-    priv->capabilities.effective |= capflag(CAP_IPC_LOCK);
-
-    priv->capabilities.permitted = priv->capabilities.effective;
+    priv->capabilities.effective = capflag(CAP_SYS_ADMIN);
     priv->capabilities.bounding = priv->capabilities.effective;
 
+    /*
+     * for some operations like container decryption, overlay mount,
+     * and creation of loop devices, the following capabilities must
+     * be in the permitted set:
+     * - CAP_MKNOD
+     * - CAP_SYS_CHROOT
+     * - CAP_SETGID
+     * - CAP_SETUID
+     * - CAP_FOWNER
+     * - CAP_DAC_OVERRIDE
+     * - CAP_DAC_READ_SEARCH
+     * - CAP_CHOWN
+     * - CAP_IPC_LOCK
```

Comment:
I don't think `CAP_IPC_LOCK` is references anywhere else here now.... missing or intended?

The encrypt flow appears to just be using `CAP_SYS_PTRACE`?

Reply:
Initially it's required to be in the permitted set, I fixed that in RPC code to request it explicitly and trigger error if missing. Also fixed cryptsetup cleanup to request CAP_SYS_ADMIN during umount